src/core/checkedint.d

```diff
@@ -415,9 +416,10 @@ unittest
 pragma(inline, true)
 long muls(long x, long y, ref bool overflow)
 {
-    long r = cast(ulong)x * cast(ulong)y;
-    enum not0or1 = ~1L;
-    if((x & not0or1) && ((r == y)? r : (r / x) != y))
+    const r = x * y;
+    if (  ~x._differentSign(y)  // x and y has the same sign
+         & r._signBit           // and result is negative (covers min * -1)
+        || x && r / x != y)     // or perform simple check for x != 0
```

Comment:
Something amazing going on with whitespace here.  Quite dizzying to review at a glance.